app/api/v1/auth/auth.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, List, Optional
+
+from datetime import datetime, timedelta
+
+from fastapi import (
+    params,
+    APIRouter,
+    Depends,
+    HTTPException,
+    Request,
+    Security,
+    status,
+)
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from jose import JWTError, jwt
+from pydantic import ValidationError
+
+from app.api.v1.models.user import UserScope
+from app.core import Settings
+from app.core.user import get_user_info
+
+from ..models import Token, TokenData
+
+from ....core import AuthenticationContext
+
+from ..tags import AUTHORIZE
+
+router = APIRouter()
+
+oauth2_scheme = OAuth2PasswordBearer(
+    tokenUrl="token",
+    scopes={
+        UserScope.admin: "The admin role can execute administrative functions.",
+        UserScope.auditor: "The auditor role is a readonly role that can observe the election",
+        UserScope.guardian: "The guardian role can excute guardian functions.",
+        UserScope.voter: "The voter role can execute voting functions such as encrypt ballot.",
+    },
+)
+
+
+class ScopedTo(params.Depends):
+    """Define a dependency on particular scope."""
+
+    def __init__(self, scopes: List[UserScope]) -> None:
+        super().__init__(self.__call__)
+        self._scopes = scopes
+
+    def __call__(
+        self,
+        request: Request,
+        settings: Settings = Settings(),
+        token: str = Security(oauth2_scheme),
+    ) -> TokenData:
+        """Check scopes and return the current user."""
+        data = validate_access_token(settings, token)
+        validate_access_token_authorization(data, self._scopes)
+        return data
+
+
+def validate_access_token_authorization(
+    token_data: TokenData, scopes: List[UserScope]
+) -> None:
+    """Validate that the access token is authorized to the requested resource."""
+    if any(scopes):
+        scope_str = ",".join(scopes)
+        authenticate_value = f'Bearer scope="{scope_str}"'
+    else:
+        authenticate_value = "Bearer"
+    for scope in scopes:
```

Comment:
this effectively makes scopes an `OR` operation so they can be "admin, or guardian, or auditor..."

Reply:
I thought we were looking at having the ability to have someone as both an admin and a guardian.

Reply:
yes. this does that. sorry wording not clear. i mean either or can access, not that they are "or exclusive"